### DISCUSSION TITLE
Welcome to Unige_stuff Discussions!

### DISCUSSION THREAD

**Discussion by ihcsof** [ihcsof/Unige_stuff#1] [Announcements]

**Comment by keatane** [comment]
### Bug fixes - DC Application
The following message relates to some bugs or inaccurate terms in the questions applications, focusing on the part about Distributed Computing (DC). 

**Question: After network partition, what hapens if the two partitions went both on and have the same number?**
_typo "happens"; "have the same number", about what, are you probably talking about Raft?_

**Question: In the context of distributed systems, what trade-off is mentioned in the text?**
_what text?_
**Question: What happens when an Propser fails in accept phase (case seen in lesson)?**
_typo "Proposer"_

**Question: Which of these isn't something to take into account for choosing the number of znodes? Performance (of r/w), Costs, Tolerable failures (avaibility)**
_I see a probable inconsistency between question and answers_

**Question: If lam = 0.9 i have**
_I've done this today, so I remember, but tomorrow I won't :), specify the context, please_

**Question: If you have CP you have:**
_Please, specify CP of CAP, in case in the future will be something called similar_

**Question: What is the key inspiration behind the design of Apache Zookeeper?**
1.  The implementation of the NFS filesystem
2.  The need for order and sequencing in data processing
3.  The coordination of distributed applications
Insert your answer (1, 2, or 3, 0 to leave): 1
Wrong answer. The correct one is: 3.
_I thought was 1, can I request a double check, please?_

**Question: What is the condition for the queue to be in equilibrium in the M/M/1 model? 1.  λ > μ**
_My congratulations to get lamda and mu printed correctly in the prompt_

**Question: You're doing great :O**
_You too_

**Question: Avg length of the queue is? 1.  1/(1-lam)**
_I've just congratulated and then this question pop up u.u_

**Question: What happens when an Propser fails in prepare phase (case seen in lesson)?**
_typo as before_

**Question: What are the three components of the CAP theorem, and what does each represent?**
_I've seen this question already three times, in different sauces, I leave you decide if remove one of these from the pool_

**Question: What is indipendente-initiated externally storing log compation?**
_typo "indipendente"_

**Comment by keatane** [comment]
# Bug fixes - FSTT Application
The following message relates to some bugs or inaccurate terms in the questions applications, focusing on the part about Function and Security Testing (FSTT).

**Question: ChromeDriver is-a ChromiumDriver is-a RemoteWebDriver: selectable with ??("webdriver.chrome.driver", <path>)**
1.  System.setProperty
2.  WebDriver = new newChromDriver()
3.  driver.get()
Insert your answer (1, 2, or 3, 0 to leave):

--> Question formally correct?

**Question: What are "close the current window, quitting the browser if it was the last" and "close the driver and all its windows"?**
1.  Selenium's quit and close
2.  close and quit
3.  none

--> answers 1 and 2 can be misunderstood

**Question: How to select b2 with XPath?**
1.  /html/body/div[1]/input[2]
2.  /html/body/div[2]
3.  //div[2]/input[2]

--> what is b2?

**Question: Maven's, config is based on ??, an XML file describing dependencies and libraries, automatically downloading them from remote repo's if not present locally**
1.  Gradle
2.  Project Object Model
3.  Document Object Model
Insert your answer (1, 2, or 3, 0 to leave): 2

Wrong answer. The correct one is: 3.

--> It is not the 2nd? Should be POM, isn't it?

**Question: To measure the ?? (how much AUT is tested by testsuite) we need to find a ?? (executed / total, e.g LOCs)**
1.  coverage, unit
2.  percentage, indicator
3.  percentage, indicator
Insert your answer (1, 2, or 3, 0 to leave): 2

Wrong answer. The correct one is: 1.

--> To be reviewed.

**Question: ?? allows multiple ways of checking for UIelements using assertions (hard) and verification (soft: the execution continues if fails)**
1.  Selenese
2.  Selesium
3.  Selenium language
Insert your answer (1, 2, or 3, 0 to leave): 3

Wrong answer. The correct one is: 1.

--> Maybe I lost something, I don't remember this. To be reviewed.

**Question: Gherkin is a ?? ?? language with # comments: a Feature keyword provides high-level description of feature and to group related scenarios**
1.  line-oriented, businnes-readable
2.  We can have multiple Scenario under a Feature
3.  both true sentences
Insert your answer (1, 2, or 3, 0 to leave): 1

Wrong answer. The correct one is: 3.

--> To be reviewed.

**Question: Docker key features are: containerization, portability, efficiency, consistency, scalability, resource optimization**
1.  Yes, all
2.  No, only first three, other are benefits
3.  No, nothing
Insert your answer (1, 2, or 3, 0 to leave):

--> subdle

**Question: Given, When, Then, And, Or, But, Background, Parameters (@Regression, @EndToEnd, ...) are ?? keywords**
1.  Gherkin
2.  Selenium
3.  Sium
Insert your answer (1, 2, or 3, 0 to leave):

--> :)

**Comment by keatane** [comment]
# Bug fixes - DECS.txt
## Bad-formatted questions
Question: In Arbitrum, what happens if a validator loses the bisection game? Their stake is partially transferred to the winner and the rest is burned or given to a verifier 
1.  Their transactions are invalidated 
2.  They are removed from the network permanently 
3.  1
Insert your answer (1, 2, or 3, 0 to leave):


Question: What is one key difference between Optimistic Rollups and Zero-Knowledge Rollups (L2 Panorama)? Zero-Knowledge Rollups rely on optimistic assumptions, while Optimistic Rollups use cryptographic proofs 
1.  Optimistic Rollups rely on challenges, while Zero-Knowledge Rollups like zk-SNARK use cryptographic proofs for correctness, being more computationally intensive 
2.  Zero-Knowledge Rollups depend on Merkle trees, while Optimistic Rollups don’t 
3.  2
Insert your answer (1, 2, or 3, 0 to leave): 


Question: Is the order of Ethereum stack right?: the EVM at its core, Ethereum nodes for blockchain state and consensus, smart contracts, client APIs for developer interaction, and standard web or mobile end-user applications 
1.  yes 
2.  no, 2 has to be exchanged with 3 
3.  no, 3 has to be exchanged with 3 
Insert your answer (1, 2, or 3, 0 to leave): 

## Wrong questions
Question: ? tokens enable multiple token types within a single contract, reducing blockchain space and gas costs, supporting batch transfers for efficient multi-token transactions, and allowing atomic swaps for trustless exchanges—features that make it ideal for applications like gaming 
1.  ERC-1155 
2.  ERC-721 
3.  ERC-20 
Insert your answer (1, 2, or 3, 0 to leave): 1

Wrong answer. The correct one is: 2
